aptos-move/e2e-move-tests/src/tests/aggregator.rs

```diff
@@ -6,160 +6,172 @@ use crate::{
-    assert_success!(outputs[13]);
-    assert_abort!(outputs[14], 25863);
-}
+    #[test]
+    #[should_panic]
+    fn test_aggregator_underflow(block_split in BlockSplit::arbitrary(3)) {
+        let (mut h, acc) = setup();
 
-#[test]
-#[should_panic]
-fn test_aggregator_underflow() {
-    let (mut h, acc) = setup();
+        let txns = vec![
 
-    let txn1 = new(&mut h, &acc, 0, 600);
-    let txn2 = add(&mut h, &acc, 0, 400);
-    let txn3 = sub(&mut h, &acc, 0, 500);
+            (0, new(&mut h, &acc, 0, 600)),
+            (0, add(&mut h, &acc, 0, 400)),
+            // Value dropped below zero - abort with EAGGREGATOR_UNDERFLOW.
+            // we cannot catch it, because we don't materialize it.
+            (0x02_0002, sub(&mut h, &acc, 0, 500)),
+        ];
 
-    assert_success!(h.run(txn1));
-    assert_success!(h.run(txn2));
+        h.run_block_in_parts_and_check(block_split, txns);
+    }
 
-    // Value dropped below zero - abort with EAGGREGATOR_UNDERFLOW.
-    assert_abort!(h.run(txn3), 131074);
-}
+    #[test]
+    fn test_aggregator_materialize_underflow(block_split in BlockSplit::arbitrary(2)) {
+        let (mut h, acc) = setup();
 
-#[test]
-fn test_aggregator_materialize_underflow() {
-    let (mut h, acc) = setup();
+        let txns = vec![
+            (0, new(&mut h, &acc, 0, 600)),
+            // Underflow on materialized value leads to abort with EAGGREGATOR_UNDERFLOW.
+            // we can catch it, because we materialize it.
+            (0x02_0002, materialize_and_sub(&mut h, &acc, 0, 400)),
+        ];
 
-    let txn1 = new(&mut h, &acc, 0, 600);
-    let txn2 = materialize_and_sub(&mut h, &acc, 0, 400);
+        h.run_block_in_parts_and_check(block_split, txns);
+    }
 
-    // Underflow on materialized value leads to abort with EAGGREGATOR_UNDERFLOW.
-    assert_success!(h.run(txn1));
-    assert_abort!(h.run(txn2), 131074);
-}
+    #[test]
+    #[should_panic]
+    fn test_aggregator_overflow(block_split in BlockSplit::arbitrary(3)) {
+        let (mut h, acc) = setup();
 
-#[test]
-#[should_panic]
-fn test_aggregator_overflow() {
-    let (mut h, acc) = setup();
+        let txns = vec![
+            (0, new(&mut h, &acc, 0, 600)),
+            (0, add(&mut h, &acc, 0, 400)),
+            // Currently, this one will panic, instead of throwing this code.
+            // we cannot catch it, because we don't materialize it.
+            (0x02_0001, add(&mut h, &acc, 0, 201)),
+        ];
 
-    let txn1 = new(&mut h, &acc, 0, 600);
-    let txn2 = add(&mut h, &acc, 0, 400);
-    let txn3 = add(&mut h, &acc, 0, 201);
+        h.run_block_in_parts_and_check(block_split, txns);
+    }
 
-    assert_success!(h.run(txn1));
-    assert_success!(h.run(txn2));
 
-    // Limit exceeded - abort with EAGGREGATOR_OVERFLOW.
-    assert_abort!(h.run(txn3), 131073);
-}
+    #[test]
+    fn test_aggregator_materialize_overflow(block_split in BlockSplit::arbitrary(2)) {
```

Comment:
This is the same as running txns one by one?

Reply:
we have two options - single block with 2 transactions, and two blocks with 1 transaction each?